src/core/configs/cluster.rs

```diff
@@ -0,0 +1,397 @@
+use itertools::Itertools;
+use std::cell::Cell;
+use std::cell::RefCell;
+use std::fs;
+
+pub trait Validatable {
+    fn validate(&self) -> Option<String>;
+
+    fn aggregate<T: Validatable>(&self, elements: &Option<Vec<T>>) -> Option<String> {
+        let options: Vec<Option<String>> = elements
+            .as_ref()?
+            .iter()
+            .map(|elem| elem.validate())
+            .filter(|f| f.is_some())
+            .collect::<Vec<Option<String>>>();
+        if options.len() > 0 {
+            return Some(
+                options
+                    .iter()
+                    .fold("".to_string(), |acc, x| acc + &x.as_ref().unwrap()),
+            );
+        }
+        None
+    }
+}
+
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+pub struct NodeDisk {
+    pub path: Option<String>,
+    pub name: Option<String>,
+}
+
+impl Validatable for NodeDisk {
+    fn validate(&self) -> Option<String> {
+        if self.path.is_none() {
+            debug!("field 'path' for 'disk' is invalid");
+            return Some("field 'path' for 'disk' is invalid".to_string());
+        }
+        if self.path.as_ref()?.is_empty() {
+            debug!("field 'path' for 'disk' is empty");
+            return Some("field 'path' for 'disk' is empty".to_string());
+        }
+        if self.name.is_none() {
+            debug!("field 'name' for 'disk' is invalid");
+            return Some("field 'name' for 'disk' is invalid".to_string());
+        }
+        if self.name.as_ref()?.is_empty() {
+            debug!("field 'name' for 'disk' is empty");
+            return Some("field 'name' for 'disk' is empty".to_string());
+        }
+        None
+    }
+}
+
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+pub struct Node {
+    pub name: Option<String>,
+    pub address: Option<String>,
+    pub disks: Option<Vec<NodeDisk>>,
+
+    #[serde(skip)]
+    pub host: RefCell<String>,
+    #[serde(skip)]
+    pub port: Cell<i32>,
+}
+
+impl Node {
+    fn prepare(&self) -> Option<String> {
+        let ip: Vec<&str> = self.address.as_ref()?.split(":").collect::<Vec<&str>>();
+        self.host.replace(ip[0].to_string());
+        let port = ip[1].parse::<i32>();
```

Comment:
If `address == "0.0.0.0" thread will panic with
`'index out of bounds: the len is 1 but the index is 1'`